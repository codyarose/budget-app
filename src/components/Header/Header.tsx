import React, { FC } from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { useDispatch, useSelector } from 'react-redux'

import { logoutUser, selectIsAuthed } from '../../features/auth/authSlice'
import ContentContainer from '../common/ContentContainer'

const Header: FC = () => {
	const dispatch = useDispatch()
	const isAuthed = useSelector(selectIsAuthed)

	const handleLogout = () => {
		dispatch(logoutUser())
	}
	return (
		<>
			{isAuthed && (
				<StyledHeader>
					<StyledContent>
						<StyledTitle to="/">
							<h1>Expenses</h1>
						</StyledTitle>
						<button onClick={handleLogout}>Logout</button>
					</StyledContent>
				</StyledHeader>
			)}
		</>
	)
}

export default Header

const StyledHeader = styled.header`
	padding: ${({ theme }) => `${theme.spacing.sm} 0`};
`
const StyledContent = styled(ContentContainer)`
	display: flex;
	justify-content: space-between;
	align-items: center;
`
const StyledTitle = styled(Link)`
	color: ${({ theme }) => theme.colors.black};
	text-decoration: none;
	h1 {
		margin: 0;
	}
`
