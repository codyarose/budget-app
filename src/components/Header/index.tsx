import React, { FC } from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'

const Header: FC = () => {
	return (
		<header>
			<h1>Budget</h1>
			<StyledNavLink exact to="/" activeClassName="is-active">
				Home
			</StyledNavLink>
			<StyledNavLink to="/create" activeClassName="is-active">
				Create
			</StyledNavLink>
		</header>
	)
}

export default Header

const StyledNavLink = styled(NavLink)`
	&.is-active {
		font-weight: 700;
	}
`
