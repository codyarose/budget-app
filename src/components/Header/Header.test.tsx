import React from 'react'
import { shallow } from 'enzyme'
import Header from './index'

describe('Header component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(<Header />)
		expect(wrapper).toMatchSnapshot()
	})
})
