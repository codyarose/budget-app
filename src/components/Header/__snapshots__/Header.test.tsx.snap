// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<header>
  <h1>
    Budget
  </h1>
  <Styled(NavLink)
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home
  </Styled(NavLink)>
  <Styled(NavLink)
    activeClassName="is-active"
    to="/create"
  >
    Create
  </Styled(NavLink)>
</header>
`;
